src/python/zquantum/core/cost_function.py

```diff
@@ -94,6 +94,11 @@ def get_ground_state_cost_function(
     Returns:
         Callable
     """
+    DeprecationWarning(
```

Comment:
You need to add `warnings.warn(DeprecationWarning(...))` in order to make it work. Same in other places.